### PR TITLE
`setup.py`: Migrate from distutils to setuptools (fixes #113)

### LINKED ISSUE #113: Replace legacy distutils by... setuptools?
Hi @ymattw,

are there any plans to migrate off legacy distutils? There are multiple options for a target, what are your thoughts and/or equirements for a pull request with a good chance of getting merged if any? 

Best, Sebastian

### PR DESCRIPTION
**Pull request by hartwork**
Fixes #113

### CODE REVIEW

File: ydiff.py [hunk 1]
```diff
@@ -28,9 +28,6 @@
                      'stdin, with side by side and auto pager support')
 }
 
-if sys.hexversion < 0x03000000:
```
Comment (ymattw) [inline ydiff.py:31]: Please keep this check, build (package) dependencies and runtime dependencies are different. Others look good. Thank you.

**Review by ymattw — CHANGES_REQUESTED** [review]

File: ydiff.py (hunk 1 above)
Comment (hartwork) [reply ydiff.py:31]: @ymattw `python_requires='>=3',` is runtime, are you sure you want me to drop that commit?

Comment (ymattw) [reply ydiff.py:31]: I meant keeping the check in ydiff.py, which can be used without installing the pypi package.

Comment (hartwork) [reply ydiff.py:31]: @ymattw I'm not sure that's worth protecting with all Pythons dead before 3.8, but can do. Pushed.

**Pull request merged**